js/shopping-list.js

```diff
@@ -0,0 +1,50 @@
+var ulElement = document.querySelector('ul');
+
+function addListItems(event){
+    var form = event.target;
+    var items = form.querySelector('#items').value.split(',');
+
+    if(items[0].length > 0){
+        document.querySelector('#user-warning').textContent = "";
+        for(var i = 0; (i < items.length) && (items[i].length > 0) && (items[i] !== ' '); i++){
+            appendItemToList(createXImgElement(), items[i]);
+        }
+    } else {
+        document.querySelector('#user-warning').textContent = "Please enter an item.";
+    }
+    event.preventDefault();
+    form.reset();
+}
+
+function createXImgElement(){
+    var imgElement = document.createElement('img');
+    imgElement.setAttribute('src', 'media/x-icon.png');
+    imgElement.setAttribute('alt', 'A red x to indicate removal.');
+    imgElement.addEventListener('click', removeListItem);
+    return imgElement;
+}
+
+function appendItemToList(imgElement, item){
+    var liElement = document.createElement('li');
+    liElement.textContent = item;
+    liElement.appendChild(imgElement);
+    ulElement.appendChild(liElement);
+}
+
+function removeListItem(event){
+    var imgElement = event.target;
+    var liElement = imgElement.parentNode;
+    ulElement.removeChild(liElement);
+}
+
+function clearList(){
+    while(ulElement.hasChildNodes()){
+        ulElement.removeChild(ulElement.firstChild);
+    }
+}
+
+document.querySelector('form').addEventListener('submit', addListItems);
+document.querySelector('button').addEventListener('click', clearList);
+
+var date = new Date();
+document.querySelector('footer').innerHTML = "Aggie Wheeler Bateman &copy; " +  date.getFullYear() + " <a href=\"https://www.aggiewheelerbateman.com\" target=\"_blank\">Personal Portfolio</a>";
```

Comment:
One possible refactor we could do for this line to make it a little easier for the next developer to understand why the `<footer>` is empty in the markup, even though it's being styled, is change the HTML and JS to the following:
![Capture](https://user-images.githubusercontent.com/20998142/84096756-8ee98780-a9b7-11ea-9bea-9804b9fef844.PNG)

[The MDN article on innerHTML](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML) mentions that use of `innerHTML` often gets rejected by automated security review software, so removing its use is generally considered good practice. Additionally, it should be more efficient to have "static" markup in the HTML instead of dynamically adding it in JS - if we use this refactor, then we're only modifying the `textContent` of a single span with the only actually dynamic value.